pkg/machinery/config/types/v1alpha1/generate/generate.go

```diff
@@ -533,3 +551,37 @@ func emptyIf(str, check string) string {
 
 	return str
 }
+
+func genV4UUID() (string, error) {
```

Comment:
do we really need this to be UUID? can we make it more compact by simply base64-encoding random bytes? we can put more randomness in the same number of bytes this way

Reply:
No, it's a completely arbitrary value.  It can be a random base64-encoded value just like the cluster secret, if preferred.

Reply:
I've just made it the same as the secret for now:  32 bytes of random data.  The length is arbitrary, though.

Reply:
Actually, it _is_ worth noting that the problem with base64 is that it isn't URL-clean.  It can contain characters which must be escaped, unlike UUIDs.

Reply:
Meh, I just used `base64.URLEncoding` instead, since the underlying bytes don't actually matter.  It's just a unique ID.